ILSpy.AddIn/CodeElementXmlDocKeyProvider.cs

```diff
@@ -0,0 +1,241 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace ICSharpCode.ILSpy.AddIn
+{
+	/// <summary>
+	/// Provides XML documentation tags for Visual Studio CodeElements.
+	/// </summary>
+	/// <remarks>
+	/// Used to support the "/navigateTo" command line option when opening ILSpy. Must match
+	/// the logic of ICSharpCode.ILSpy.XmlDoc.XmlDocKeyProvider, which does the same thing for
+	/// a Mono.Cecil.MemberReference. See "ID string format" in Appendix A of the C# language
+	/// specification for formatting requirements.
+	/// </remarks>
+	public static class CodeElementXmlDocKeyProvider
+	{
+		#region GetKey
+		public static string GetKey(EnvDTE.CodeElement member)
```

Comment:
NOTE: I initially attempted to follow the structural logic of XmlDocKeyProvider, but found that EnvDTE.CodeElement and friends don't precisely match their Mono.Cecil counterparts, and the end an approach based mainly on string parsing (of type FullNames) proved much simpler.